File: artifacts/tulipindicators/decompose/src/sqrt.c
int ti_sqrt_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_sqrt(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *in1 = inputs[0];
  (void) options;
  double *output = outputs[0];
  int i;
  for (i = 0; i < size; i += 1)
  {
    output[i] = sqrt(in1[i]);
  }

  return 0;
}


/*** DEPENDENCIES:
***/

File: artifacts/tulipindicators/translate/src/sqrt.rs
use crate::*;
use libm::sqrt;

pub fn ti_sqrt_start(options: Option<&f64>) -> i32 {
    let _ = options; // Explicitly ignore the parameter
    0
}

pub fn ti_sqrt(
    size: i32,
    inputs: &[&[f64]],
    options: &[f64],
    outputs: &mut [&mut [f64]],
) -> i32 {
    let in1 = inputs[0];
    let output = &mut outputs[0];
    
    for i in 0..size as usize {
        output[i] = sqrt(in1[i]);
    }
    
    0
}
